xarray/plot/plot.py

```diff
@@ -326,40 +337,127 @@ def plot(
+                "Only 1d and 2d plots are supported for facets in xarray. "
+                "See the package `Seaborn` for more options."
+            )
         plotfunc = hist
 
     kwargs["ax"] = ax
 
     return plotfunc(darray, **kwargs)
 
 
+@overload
+def line(
+    darray,
+    *args: Any,
+    row: Hashable,
+    col: Hashable | None = None,
+    figsize: Iterable[float] | None = None,
+    aspect: float | None = None,
+    size: float | None = None,
+    ax: Axes | None = None,
+    hue: Hashable | None = None,
+    x: Hashable | None = None,
+    y: Hashable | None = None,
+    xincrease: bool | None = None,
+    yincrease: bool | None = None,
+    xscale: MPLScaleOptions = None,
+    yscale: MPLScaleOptions = None,
+    xticks: ArrayLike | None = None,
+    yticks: ArrayLike | None = None,
+    xlim: ArrayLike | None = None,
+    ylim: ArrayLike | None = None,
+    add_legend: bool = True,
+    _labels: bool = True,
+    **kwargs: Any,
+) -> FacetGrid:
+    ...
+
+
+@overload
+def line(
+    darray,
+    *args: Any,
+    row: Hashable | None = None,
+    col: Hashable,
+    figsize: Iterable[float] | None = None,
+    aspect: float | None = None,
+    size: float | None = None,
+    ax: Axes | None = None,
+    hue: Hashable | None = None,
+    x: Hashable | None = None,
+    y: Hashable | None = None,
+    xincrease: bool | None = None,
+    yincrease: bool | None = None,
+    xscale: MPLScaleOptions = None,
+    yscale: MPLScaleOptions = None,
+    xticks: ArrayLike | None = None,
+    yticks: ArrayLike | None = None,
+    xlim: ArrayLike | None = None,
+    ylim: ArrayLike | None = None,
+    add_legend: bool = True,
+    _labels: bool = True,
+    **kwargs: Any,
+) -> FacetGrid:
+    ...
+
+
+@overload
+def line(
+    darray: DataArray,
+    *args: Any,
+    row: None = None,
+    col: None = None,
+    figsize: Iterable[float] | None = None,
+    aspect: float | None = None,
+    size: float | None = None,
+    ax: Axes | None = None,
+    hue: Hashable | None = None,
+    x: Hashable | None = None,
+    y: Hashable | None = None,
+    xincrease: bool | None = None,
+    yincrease: bool | None = None,
+    xscale: MPLScaleOptions = None,
+    yscale: MPLScaleOptions = None,
+    xticks: ArrayLike | None = None,
+    yticks: ArrayLike | None = None,
+    xlim: ArrayLike | None = None,
+    ylim: ArrayLike | None = None,
+    add_legend: bool = True,
+    _labels: bool = True,
+    **kwargs: Any,
+) -> list[Line3D]:
+    ...
+
+
 # This function signature should not change so that it can use
 # matplotlib format strings
 def line(
-    darray,
-    *args,
-    row=None,
-    col=None,
-    figsize=None,
-    aspect=None,
-    size=None,
-    ax=None,
-    hue=None,
-    x=None,
-    y=None,
-    xincrease=None,
-    yincrease=None,
-    xscale=None,
-    yscale=None,
-    xticks=None,
-    yticks=None,
-    xlim=None,
-    ylim=None,
-    add_legend=True,
-    _labels=True,
-    **kwargs,
-):
+    darray: DataArray,
```

Comment:
Is it now common to also type the real function when having overloads?

Reply:
I guess if your goal is to have typing support for people using the function it is useless.
But it helps if you want to typecheck the function :)

Reply:
Also, in modern python the type annotations are stored in the signature, but only of the "real" function and not the overloads.